src/sidebar/views/overview.ts

```diff
@@ -0,0 +1,134 @@
+import { Command, Event, EventEmitter, ExtensionContext, TreeDataProvider, TreeItem, TreeView, window } from 'vscode';
+import { ExtensionApi } from '../../backend/api';
+import { CheckerMetadata } from '../../backend/types';
+
+export class OverviewItem {
+    constructor(private label: string | (() => string), private command?: Command) {}
+
+    getTreeItem(): TreeItem | Promise<TreeItem> {
+        const label = typeof this.label === 'string' ? this.label : this.label();
+
+        const node = new TreeItem(label);
+        node.command = this.command;
+        node.description = this.command?.tooltip;
+
+        return node;
+    }
+}
+
+export class OverviewView implements TreeDataProvider<string> {
+    private tree?: TreeView<string>;
+
+    // FIXME: Export the keys into an enum, or otherwise check in the itemsList if an entry exists
+    private items: {[id: string]: OverviewItem} = {
+        'loading': new OverviewItem('Loading overview, please wait...'),
+
+        'notfound': new OverviewItem('CodeChecker run not found.'),
+        'notfound2': new OverviewItem('Run CodeChecker, reload metadata, or set the output folder to get started'),
+
+        'files': new OverviewItem(() => 'Files analyzed: ' + ExtensionApi.metadata.sourceFiles.size),
```

Comment:
We can use string interpolation to simplify the code:
```suggestion
        'files': new OverviewItem(() => `Files analyzed: ${ExtensionApi.metadata.sourceFiles.size}`),
```
Please use it in other places as well.